### PR TITLE
IfcReferent nesting

### LINKED ISSUE #112: `IfcReferent` nesting documentation
The `IfcReferent` shall be nested using its own `IfcRelNests` and not reuse the one that nests `IfcAlignmentHorizontal` and other parts.

EDIT: I see that the documentation is not clear about this. Will amend accordingly.

_Originally posted by @pjanck in https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/issues/109#issuecomment-1488454286_

### PR DESCRIPTION
**Pull request by pjanck**
## Miscellaneous

fixes #112

### CODE REVIEW

File: sprints/sprint2_1.md [hunk 1]
```diff
@@ -38,6 +38,6 @@ This sprint encompasses the following usages:
     - There shall be at least one `CONSTANTGRADIENT` and either at least one `CIRCULARARC` or at least one `PARABOLICARC` segments nested in the vertical layout (see `IfcAlignmentVerticalSegment.PredefinedType`).
 - [Object Nesting](https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/object-nesting.htm)
     - There shall be at least one `IfcReferent` on `IfcAlignment` with non-empty `Name`.
-    - All `IfcReferent` shall be nested in any `IfcAlignment`.
+    - All `IfcReferent` shall be nested in any `IfcAlignment`. The `IfcRelNests` relationship shall only nest `IfcReferent` instances (see https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/issues/112).
```
Comment (SergejMuhic) [inline sprints/sprint2_1.md:41]: Are we sure about this? We also have to consider Alignment Layout.

Comment (pjanck) [reply sprints/sprint2_1.md:41]: Alignment layout uses another instance of `IfcRelNests`. Perhaps we need to reword a bit?

Comment (SergejMuhic) [reply sprints/sprint2_1.md:41]: Just adding something like "The `IfcRelNests` relationship nesting `IfcReferent` occurrences shall only nest instances of `IfcReferent` (see https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/issues/112)."

Comment (larswik) [reply sprints/sprint2_1.md]: Where is this adressed in the documentation?

**Review by larswik — APPROVED** [review]
I am sort of missing a firm requirement that IfcReferent nesting and IfcLinearElement nesting should be in different IfcRelNests instances.

**Comment by SergejMuhic** [comment]
Usages should have their own tree to traverse. Especially with nesting which has a strict order of elements.

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**